internal/gomod/version.go

```diff
@@ -69,99 +69,117 @@ func GetModuleVersion(moduleDir string) (string, error) {
 	}
 
-	latestTag, err := GetLatestTag(repo, headCommit)
-	if err != nil {
-		if errors.Is(err, plumbing.ErrObjectNotFound) {
-			return module.PseudoVersion("v0", "", headCommit.Author.When, headCommit.Hash.String()[:12]), nil
-		}
-
-		return "", err
-	}
-
-	if latestTag.commit.Hash.String() == headCommit.Hash.String() {
-		return latestTag.name, nil
+	if string(latestTagCommit) == string(headCommit) {
+		return string(latestTag), nil
 	}
 
-	return module.PseudoVersion(
-		semver.Major(latestTag.name),
-		latestTag.name,
-		latestTag.commit.Author.When,
-		latestTag.commit.Hash.String()[:12],
-	), nil
-}
-
-type tag struct {
-	name   string
-	commit *object.Commit
-}
-
-// GetLatestTag determines the latest tag relative to HEAD.
-// Only tags with valid semver are considered.
-func GetLatestTag(repo *git.Repository, headCommit *object.Commit) (*tag, error) {
-	log.Debug().
-		Str("headCommit", headCommit.Hash.String()).
-		Msg("getting latest tag for head commit")
-
-	tagRefs, err := repo.Tags()
+	when, err := doExec("git", []string{"show", "--show-signature", strings.TrimSpace(string(latestTagCommit)), "--format=%cd", "--date=format:%Y-%m-%d %H:%M:%S"}, moduleDir, nil)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	var latestTag tag
-
-	err = tagRefs.ForEach(func(ref *plumbing.Reference) error {
-		if semver.IsValid(ref.Name().Short()) {
-			rev := plumbing.Revision(ref.Name().String())
-
-			commitHash, err := repo.ResolveRevision(rev)
-			if err != nil {
-				return err
-			}
-
-			commit, err := repo.CommitObject(*commitHash)
-			if err != nil {
-				return err
-			}
-
-			isBeforeOrAtHead := commit.Committer.When.Before(headCommit.Author.When) ||
-				commit.Committer.When.Equal(headCommit.Committer.When)
-
-			if isBeforeOrAtHead && (latestTag.commit == nil || commit.Committer.When.After(latestTag.commit.Committer.When)) {
-				latestTag.name = ref.Name().Short()
-				latestTag.commit = commit
-			}
-		} else {
-			log.Debug().
-				Str("tag", ref.Name().Short()).
-				Str("hash", ref.Hash().String()).
-				Str("reason", "not a valid semver").
-				Msg("skipping tag")
+	if strings.Contains(string(when), "gpg: Can't check signature: No public key") {
+		when, err = doExec("sh", []string{"-c", "tail -n +4 | head -n1"}, moduleDir, bytes.NewBuffer(when))
+		if err != nil {
+			return "", err
 		}
+	}
 
-		return nil
-	})
+	suffix := strings.TrimSpace(string(when))
+	pseudoTime, err := time.Parse("2006-01-02 15:04:05", suffix)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	if latestTag.commit == nil {
-		return nil, plumbing.ErrObjectNotFound
-	}
+	return module.PseudoVersion(
+		semver.Major(strings.TrimSpace(string(latestTag))),
+		strings.TrimSpace(string(latestTag)),
+		pseudoTime,
+		strings.TrimSpace(string(latestTagCommit))[:12],
+	), nil
+}
 
-	return &latestTag, nil
+func doExec(cmd string, args []string, workDir string, stdin io.Reader) ([]byte, error) {
+	c := exec.Command(cmd, args...)
```

Comment:
Executed commands should be logged with `DEBUG` level, see https://github.com/CycloneDX/cyclonedx-gomod/blob/v1.1.0/internal/gocmd/gocmd.go#L172-L175 for example.